### ISSUE TITLE
Two Analyzer warnings: "Function call argument is an uninitialized value (within a call to)"

### ISSUE THREAD

**Issue by sreejithp** [itsbonczek/kingpin#69]
I'm getting two logical error report from kingpin, while i do analysis on the project. I'm not sure about what is going on. Could you please take a look and fix it if you can?

![screen shot 2014-10-24 at 7 33 16 pm](https://cloud.githubusercontent.com/assets/525291/4770497/76ceac96-5b86-11e4-856b-20bdd84495c2.png)

PS:Thanks for the effort, I'm using it in one my project.You reduced much of my effort :)

**Comment by akhilcb** [comment]
I am getting same warning with Xcode 7 beta too. Did anyone get any solution for this?

**Comment by stanislaw** [comment]
@akhilcb, I'll review this again and let you know within two days.

**Comment by stanislaw** [comment]
I have looked at both of these warnings and here is my summary which is both good and bad news:

Good news:

Both warnings are similar and I was able to reproduce the warning from KPGridClusteringAlgorithm on a more simple code in [Demo project](https://github.com/stanislaw/Reports/tree/20150615-xcode-analyzer-bug) which makes it pretty obvious that it is the bug in analyzer.

I opened radar rdar://21383256 about it: [Xcode Analyzer produces wrong warning: "Function call argument is an uninitialized value"](http://www.openradar.appspot.com/21383256).

Bad news:

I have not been able yet to tweak the Kingpin's code to make analyzer happy. 

I understand that it is A MUST for external library to compile without any warnings so I need more time to understand how to accomplish that.

**Comment by stanislaw** [comment]
@akhilcb, I have updated podspec to 0.2.5. Try kingpin without these warnings.

It is impossible to wait when that radar will be considered by Apple this is why I fooled Analyzer in one of the warnings (that hack is another proof of presence of bug in analyzer) and completely suppressed the second warning. If interested see https://github.com/itsbonczek/kingpin/commit/5d96ef96e2a8722ae0c4d3267873e60a18d3a4d8 fo details.
